From 3939ed26560466c8fc90f7b07ba227729dff86b1 Mon Sep 17 00:00:00 2001
From: Dirk Neukirchen <dirkneukirchen@web.de>
Date: Wed, 6 May 2015 19:51:27 +0200
Subject: [PATCH 05/16] fuse: fix build on arm64

error:
../include/fuse_kernel.h:93:15: error: conflicting types for 'int64_t'
../include/fuse_kernel.h:92:15: error: conflicting types for 'uint64_t'

reported by buildbot

fix by adopting upstream patch:
http://git.baserock.org/cgi-bin/cgit.cgi/delta/fuse.git/commit/include/fuse_kernel.h?id=914871b20a901e3e1e981c92bc42b1c93b7ab81b

Signed-off-by: Dirk Neukirchen <dirkneukirchen@web.de>
---
 ...ckport_arm64_fuse_kernel_h_clean_includes.patch | 34 ++++++++++++++++++++++
 1 file changed, 34 insertions(+)
 create mode 100644 package/utils/fuse/patches/200-backport_arm64_fuse_kernel_h_clean_includes.patch

diff --git a/package/utils/fuse/patches/200-backport_arm64_fuse_kernel_h_clean_includes.patch b/package/utils/fuse/patches/200-backport_arm64_fuse_kernel_h_clean_includes.patch
new file mode 100644
index 0000000..8436224
--- /dev/null
+++ b/package/utils/fuse/patches/200-backport_arm64_fuse_kernel_h_clean_includes.patch
@@ -0,0 +1,34 @@
+From 914871b20a901e3e1e981c92bc42b1c93b7ab81b Mon Sep 17 00:00:00 2001
+From: Riku Voipio <riku.voipio@linaro.org>
+Date: Thu, 07 Feb 2013 11:04:21 +0000
+Subject: fuse_kernel.h: clean includes
+
+Use <linux/types.h> for linux and define types used for other operating systems
+using <stdint.h> types.
+---
+(limited to 'include/fuse_kernel.h')
+
+diff --git a/include/fuse_kernel.h b/include/fuse_kernel.h
+index 501450c..df8e9b9 100644
+--- a/include/fuse_kernel.h
++++ b/include/fuse_kernel.h
+@@ -91,12 +91,16 @@
+ #ifndef _LINUX_FUSE_H
+ #define _LINUX_FUSE_H
+ 
+-#include <sys/types.h>
++#ifdef __linux__
++#include <linux/types.h>
++#else
++#include <stdint.h>
+ #define __u64 uint64_t
+ #define __s64 int64_t
+ #define __u32 uint32_t
+ #define __s32 int32_t
+ #define __u16 uint16_t
++#endif
+ 
+ /*
+  * Version negotiation:
+--
+cgit v0.9.0.3-67-gacbf
-- 
2.4.0

